program/src/instruction.rs

```diff
@@ -79,10 +115,11 @@ pub(crate) fn decode_instruction_type(input: &[u8]) -> Result<SlashingInstructio
 
 /// Utility function for decoding instruction data
 pub(crate) fn decode_instruction_data<T: Pod>(input_with_type: &[u8]) -> Result<&T, ProgramError> {
-    if input_with_type.len() != pod_get_packed_len::<T>().saturating_add(1) {
+    let data_len = pod_get_packed_len::<T>().saturating_add(1);
+    if input_with_type.len() < data_len {
```

Comment:
Changed this to `<` to allow storage of the sigverify data in slashing instruction.